openstack/blockstorage/extensions/backups/requests.go

```diff
@@ -182,3 +183,67 @@ func Update(client *gophercloud.ServiceClient, id string, opts UpdateOptsBuilder
 	})
 	return
 }
+
+// CreateVolumeExt is the structure used when creating new volume from a backup
+// resource. It embeds volumes.CreateOpts and so inherits all of its required
+// and optional fields, with the addition of the backup_id field.
+type CreateVolumeExt struct {
+	volumes.CreateOptsBuilder
+	// The ID of the volume backup from which you want to create the volume.
+	// Supported since microversion 3.47
+	BackupID string `json:"backup_id,omitempty"`
+}
+
+// ToVolumeCreateMap adds the backup_id option to the base volume creation
+// options.
+func (opts CreateVolumeExt) ToVolumeCreateMap() (map[string]interface{}, error) {
+	base, err := opts.CreateOptsBuilder.ToVolumeCreateMap()
+	if err != nil {
+		return nil, err
+	}
+
+	if opts.BackupID == "" {
+		return base, nil
+	}
+
+	volumeMap := base["volume"].(map[string]interface{})
+	volumeMap["backup_id"] = opts.BackupID
+
+	// delete size, when restoring from a backup
+	if v, ok := volumeMap["size"].(float64); ok && v == 0 {
+		delete(volumeMap, "size")
+	}
+
+	return base, nil
+}
+
+// RestoreOpts contains options for restoring a Backup. This object is passed to
+// the backups.Restore function.
+type RestoreOpts struct {
+	// VolumeID is the ID of the existing volume to restore the backup to.
+	VolumeID string `json:"volume_id,omitempty"`
+
+	// Name is the name of the new volume to restore the backup to.
+	Name string `json:"name,omitempty"`
+}
+
+// ToBackupRestoreMap assembles a request body based on the contents of a
+// RestoreOpts.
+func (opts RestoreOpts) ToBackupRestoreMap() (map[string]interface{}, error) {
+	return gophercloud.BuildRequestBody(opts, "restore")
+}
+
+// Restore will restore a Backup to a volume based on the values in RestoreOpts. To
+// extract the Restore object from the response, call the Extract method on the
+// RestoreResult.
+func BackupRestore(client *gophercloud.ServiceClient, id string, opts RestoreOpts) (r RestoreResult) {
```

Comment:
Let's change this to just `Restore`. Since all of this is namespaced under a "backup" package, having `Backup` in the function name is redundant.

```
backups.Restore(client, "uuid", restoreOpts)
```

Reply:
I used `BackupRestore`, because the `Restore` struct already has this name.

Reply:
Oh right - of course.

What about `RestoreFromBackup` for the function name? Or can the result struct be renamed to something else? `BackupRestore` just feels a little awkward - I realize this is very nit-picky, though.

Reply:
`RestoreFromBackup` looks fine.